x-pack/plugins/alerting/server/rules_client/lib/retry_if_bulk_conflicts.ts

```diff
@@ -16,50 +16,36 @@ import { RawRule } from '../../types';
 
 const MAX_RULES_IDS_IN_RETRY = 1000;
 
-export type BulkEnableOperation = (filter: KueryNode | null) => Promise<{
+interface BulkOperationReturn {
   errors: BulkOperationError[];
   rules: Array<SavedObjectsBulkUpdateObject<RawRule>>;
-  taskIdsToEnable: string[];
-}>;
-
-interface ReturnRetry {
-  errors: BulkOperationError[];
-  rules: Array<SavedObjectsBulkUpdateObject<RawRule>>;
-  taskIdsToEnable: string[];
+  accList: string[][];
 }
 
-/**
- * Retries BulkEnable requests
- * If in response are presents conflicted savedObjects(409 statusCode), this util constructs filter with failed SO ids and retries bulkEnable operation until
- * all SO updated or number of retries exceeded
- * @param logger
- * @param bulkEnableOperation
- * @param filter - KueryNode filter
- * @param retries - number of retries left
- * @param accErrors - accumulated conflict errors
- * @param accRules - accumulated enabled rules
- * @param accTaskIdsToEnable - accumulated task ids
- * @returns Promise<ReturnRetry>
- */
-
-export const retryIfBulkEnableConflicts = async (
+export const retryIfBulkConflicts = async (
+  action: 'DELETE' | 'ENABLE' | 'DISABLE',
   logger: Logger,
-  bulkEnableOperation: BulkEnableOperation,
+  bulkOperation: (filter: KueryNode | null) => Promise<BulkOperationReturn>,
```

Comment:
It seems a bit redundant that we need to pass in both the `action` and `bulkOperation` parameters. I think we should simply the params to exclude the `bulkOperation` property, since we should be able to infer the operation from the action. Having both makes the API a bit messy

Reply:
@JiaweiWu I see advantages and disadvantages of this approach.
If all bulk operations looked like: `(filterKueryNode: KueryNode | null) => this.bulkEnableRulesWithOCC({ filter: filterKueryNode }),`, for sure it's better to have some object which connects action and bulkOperation.
But we have bulkEdit, which looks like:
`(filterKueryNode: KueryNode | null) =>
        this.bulkEditOcc({
          filter: filterKueryNode,
          operations: options.operations,
          paramsModifier: options.paramsModifier,
        }),`
And  I'll need to pass more params, which also will make everything more messy. And what if other bulk operations also will need more params.
What do you think?